### ISSUE TITLE
TiKV aborts when accesses hardware info

### ISSUE THREAD

**Issue by youjiali1995** [tikv/tikv#9555]
### What version of TiKV are you using?

v4.0.10

### What operating system and CPU are you using?

https://asktug.com/t/topic/68102/3. Not known but deployed using docker.

### Steps to reproduce

Access the diagnostic service in TiKV to get the hardware info.

### What did you expect?
TiKV works well.

### What did happened?
TiKV aborts due to stack overflow:
```
(gdb) bt

#0  0x00007fa500e4162f in abort () from /usr/glibc-compat/lib/libc.so.6

#1  0x000055a5dd164c97 in std::sys::unix::abort_internal () at src/libstd/sys/unix/mod.rs:155

#2  0x000055a5dd164c85 in std::sys_common::util::abort () at src/libstd/sys_common/util.rs:20

#3  0x000055a5dd17c7d3 in std::sys::unix::stack_overflow::imp::signal_handler () at src/libstd/sys/unix/stack_overflow.rs:99

#4  <signal handler called>

#5  0x000055a5dd78a16e in extent_split_impl (tsdn=tsdn@entry=0x7fa458bfcbd0, arena=0x7fa4fd6008c0, r_extent_hooks=0x7fa458a00790, extent=0x7fa4fd60e700, size_a=size_a@entry=4096, szind_a=szind_a@entry=2, slab_a=slab_a@entry=true,  
    size_b=size_b@entry=8192, szind_b=szind_b@entry=232, slab_b=slab_b@entry=false, growing_retained=true) at src/extent.c:2103

#6  0x000055a5dd78b26a in extent_split_interior (tsdn=tsdn@entry=0x7fa458bfcbd0, arena=arena@entry=0x7fa4fd6008c0, r_extent_hooks=r_extent_hooks@entry=0x7fa458a00790, rtree_ctx=rtree_ctx@entry=0x7fa458bfcc00,  
    extent=extent@entry=0x7fa458a00420, lead=lead@entry=0x7fa458a00430, trail=trail@entry=0x7fa458a00440, to_leak=to_leak@entry=0x7fa458a00450, to_salvage=to_salvage@entry=0x7fa458a00460, size=size@entry=4096, pad=pad@entry=0,  
    alignment=alignment@entry=4096, slab=slab@entry=true, szind=szind@entry=2, growing_retained=true, new_addr=0x0) at src/extent.c:1017

#7  0x000055a5dd78e622 in extent_recycle_split (new_addr=0x0, growing_retained=true, extent=0x7fa4fd60e700, szind=2, slab=true, alignment=4096, pad=0, size=4096, extents=0x7fa4fd605878, rtree_ctx=0x7fa458bfcc00,  
    r_extent_hooks=0x7fa458a00790, arena=0x7fa4fd6008c0, tsdn=0x7fa458bfcbd0) at src/extent.c:1066

#8  extent_recycle (tsdn=tsdn@entry=0x7fa458bfcbd0, arena=arena@entry=0x7fa4fd6008c0, r_extent_hooks=r_extent_hooks@entry=0x7fa458a00790, extents=extents@entry=0x7fa4fd605878, new_addr=new_addr@entry=0x0, size=<optimized out>,  
    pad=pad@entry=0, alignment=alignment@entry=4096, slab=slab@entry=true, szind=szind@entry=2, zero=zero@entry=0x7fa458a00770, commit=commit@entry=0x7fa458a00780, growing_retained=true) at src/extent.c:1148

#9  0x000055a5dd78ff8a in extent_alloc_retained (commit=<optimized out>, zero=<optimized out>, szind=<optimized out>, slab=<optimized out>, alignment=<optimized out>, pad=<optimized out>, size=<optimized out>,  
    new_addr=<optimized out>, r_extent_hooks=<optimized out>, arena=0x7fa4fd6008c0, tsdn=0x7fa458bfcbd0) at src/extent.c:1471

#10 _rjem_je_extent_alloc_wrapper (tsdn=tsdn@entry=0x7fa458bfcbd0, arena=arena@entry=0x7fa4fd6008c0, r_extent_hooks=r_extent_hooks@entry=0x7fa458a00790, new_addr=new_addr@entry=0x0, size=<optimized out>, pad=pad@entry=0,  
    alignment=alignment@entry=4096, slab=slab@entry=true, szind=szind@entry=2, zero=zero@entry=0x7fa458a00770, commit=commit@entry=0x7fa458a00780) at src/extent.c:1539

#11 0x000055a5dd7696c7 in arena_slab_alloc_hard (bin_info=<optimized out>, szind=2, r_extent_hooks=0x7fa458a00790, arena=0x7fa4fd6008c0, tsdn=0x0) at src/arena.c:1217

#12 arena_slab_alloc (bin_info=0x55a5de6e4b30 <_rjem_je_bin_infos+80>, binshard=0, binind=0, arena=0x7fa4fd6008c0, tsdn=0x0) at src/arena.c:1247

#13 arena_bin_nonfull_slab_get (binshard=<optimized out>, binind=0, bin=0x7fa4fd607e28, arena=0x7fa4fd6008c0, tsdn=0x0) at src/arena.c:1283

#14 arena_bin_malloc_hard (tsdn=tsdn@entry=0x7fa458bfcbd0, arena=arena@entry=0x7fa4fd6008c0, bin=bin@entry=0x7fa4fd607e28, binind=binind@entry=2, binshard=binshard@entry=0) at src/arena.c:1319

#15 0x000055a5dd76c878 in _rjem_je_arena_tcache_fill_small (tsdn=<optimized out>, arena=<optimized out>, tcache=<optimized out>, tbin=<optimized out>, binind=<optimized out>, prof_accumbytes=<optimized out>) at src/arena.c:1407

#16 0x000055a5dd7c21ae in _rjem_je_tcache_alloc_small_hard (tsdn=tsdn@entry=0x7fa458bfcbd0, arena=<optimized out>, tcache=tcache@entry=0x7fa458bfcdc0, tbin=tbin@entry=0x7fa458bfce00, binind=<optimized out>,  
    tcache_success=tcache_success@entry=0x7fa458a008f0) at src/tcache.c:94

#17 0x000055a5dd7597c5 in tcache_alloc_small (size=<optimized out>, slow_path=false, zero=false, binind=2, tcache=0x7fa458bfcdc0, arena=<optimized out>, tsd=0x7fa458bfcbd0) at include/jemalloc/internal/tcache_inlines.h:60

#18 arena_malloc (slow_path=false, tcache=0x7fa458bfcdc0, zero=false, ind=2, size=<optimized out>, arena=0x0, tsdn=0x7fa458bfcbd0) at include/jemalloc/internal/arena_inlines_b.h:165

#19 iallocztm (slow_path=false, arena=0x0, is_internal=false, tcache=0x7fa458bfcdc0, zero=false, ind=2, size=<optimized out>, tsdn=0x7fa458bfcbd0) at include/jemalloc/internal/jemalloc_internal_inlines_c.h:53

#20 imalloc_no_sample (sopts=<optimized out>, ind=2, usize=32, size=<optimized out>, tsd=0x7fa458bfcbd0, dopts=<synthetic pointer>) at src/jemalloc.c:1949

#21 imalloc_body (tsd=0x7fa458bfcbd0, dopts=<synthetic pointer>, sopts=<synthetic pointer>) at src/jemalloc.c:2149

#22 imalloc (dopts=<synthetic pointer>, sopts=<synthetic pointer>) at src/jemalloc.c:2258

#23 _rjem_je_malloc_default (size=<optimized out>) at src/jemalloc.c:2289

#24 0x000055a5dd168886 in <tikv_jemallocator::Jemalloc as core::alloc::GlobalAlloc>::alloc (layout=..., self=<optimized out>) at /rust/registry/src/github.com-1ecc6299db9ec823/tikv-jemallocator-0.4.0/src/lib.rs:101

#25 __rg_alloc (arg0=<optimized out>, arg1=<optimized out>) at /home/jenkins/agent/workspace/build_tikv_multi_branch_v4.0.10/tikv/components/tikv_alloc/src/lib.rs:113

#26 alloc::alloc::alloc () at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/liballoc/alloc.rs:84

#27 <alloc::alloc::Global as core::alloc::Alloc>::alloc () at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/liballoc/alloc.rs:172

#28 alloc::raw_vec::RawVec<T,A>::allocate_in () at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/liballoc/raw_vec.rs:98

#29 alloc::raw_vec::RawVec<T>::with_capacity () at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/liballoc/raw_vec.rs:167

#30 alloc::vec::Vec<T>::with_capacity () at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/liballoc/vec.rs:359

#31 <&str as std::ffi::c_str::CString::new::SpecIntoVec>::into_vec () at src/libstd/ffi/c_str.rs:348

#32 std::ffi::c_str::CString::new () at src/libstd/ffi/c_str.rs:354

#33 0x000055a5dd17504b in std::sys::unix::fs::canonicalize () at src/libstd/sys/unix/fs.rs:1034

#34 0x000055a5dd1937d0 in sysinfo::linux::disk::find_type_for_name (name=<optimized out>) at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libstd/fs.rs:1866

#35 0x000055a5dd1939d2 in sysinfo::linux::disk::find_type_for_name (name=<optimized out>) at /rust/registry/src/github.com-1ecc6299db9ec823/sysinfo-0.14.1/src/linux/disk.rs:117

#36 0x000055a5dd1939d2 in sysinfo::linux::disk::find_type_for_name (name=<optimized out>) at /rust/registry/src/github.com-1ecc6299db9ec823/sysinfo-0.14.1/src/linux/disk.rs:117

#37 0x000055a5dd1939d2 in sysinfo::linux::disk::find_type_for_name (name=<optimized out>) at /rust/registry/src/github.com-1ecc6299db9ec823/sysinfo-0.14.1/src/linux/disk.rs:117

...

#7628 0x000055a5dd1939d2 in sysinfo::linux::disk::find_type_for_name (name=<optimized out>) at /rust/registry/src/github.com-1ecc6299db9ec823/sysinfo-0.14.1/src/linux/disk.rs:117

#7629 0x000055a5dd1939d2 in sysinfo::linux::disk::find_type_for_name (name=<optimized out>) at /rust/registry/src/github.com-1ecc6299db9ec823/sysinfo-0.14.1/src/linux/disk.rs:117

#7630 0x000055a5dd193209 in <core::iter::adapters::FilterMap<I,F> as core::iter::traits::iterator::Iterator>::next (self=0x7fa458bf9380) at /rust/registry/src/github.com-1ecc6299db9ec823/sysinfo-0.14.1/src/linux/disk.rs:78

#7631 0x000055a5dd1917a4 in <sysinfo::linux::system::System as sysinfo::traits::SystemExt>::refresh_disks_list (self=0x55a5de6a01f0 <<tikv_util::sys::SYS_INFO as core::ops::deref::Deref>::deref::__stability::LAZY+16>)

    at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/liballoc/vec.rs:2115

#7632 0x000055a5dd5abee5 in tikv::server::service::diagnostics::sys::hardware_info (collector=0x7fa458bf97f0) at src/server/service/diagnostics/sys.rs:331

#7633 0x000055a5dd5a6ef9 in <futures_cpupool::MySender<F,core::result::Result<<F as futures::future::Future>::Item,<F as futures::future::Future>::Error>> as futures::future::Future>::poll (self=<optimized out>)

    at src/server/service/diagnostics/mod.rs:143

#7634 0x000055a5dcb63b9e in std::sys_common::backtrace::__rust_begin_short_backtrace (f=...) at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/future/mod.rs:113

#7635 0x000055a5dcb62721 in core::ops::function::FnOnce::call_once{{vtable-shim}} () at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/libstd/thread/mod.rs:469

#7636 0x000055a5dd17b42f in <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once () at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/liballoc/boxed.rs:1022

#7637 0x000055a5dd17de70 in <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once () at /rustc/0de96d37fbcc54978458c18f5067cd9817669bc8/src/liballoc/boxed.rs:1022

#7638 std::sys_common::thread::start_thread () at src/libstd/sys_common/thread.rs:13

#7639 std::sys::unix::thread::Thread::new::thread_start () at src/libstd/sys/unix/thread.rs:80

#7640 0
```

**Comment by breeswish** [comment]
Should be fixed by #8796 .

**Comment by MimeLyc** [comment]
update
